libraries/engage/core/hooks/usePageConfig.js

```diff
@@ -1,30 +1,13 @@
-import defaultsDeep from 'lodash/defaultsDeep';
-import { themeConfig } from '@shopgate/pwa-common/helpers/config';
 import { useRoute } from './useRoute';
-import { getThemeSettings } from '../config/getThemeSettings';
+import { getPageConfig } from '../config/getPageConfig';
 
 /**
- * Retrieves the configuration for the current page.
+ * Retrieves the pure configuration for the current page.
+ * See documentation of `getPageConfig` for further details.
+ *
  * @returns {Object}
  */
 export function usePageConfig() {
-  const route = useRoute();
-  const { pages } = themeConfig;
-  const globalSettings = getThemeSettings();
-  const page = pages.find(element => element.pattern === route.pattern);
-
-  if (!page) {
-    return {};
-  }
-
-  const {
-    name, id, pattern, ...config
-  } = page;
-
-  const settings = defaultsDeep(config.settings, globalSettings);
-
-  return {
-    ...config,
-    settings,
-  };
+  const { pattern: pagePattern } = useRoute();
```

Comment:
This is not needed if the `useCurrentRoute()` helper is used inside `getPageConfig()` as mentioned above.

Reply:
This was done on purpose to keep the core functions pure (no magic, no dependencies).
The hook is exactly there to tie the context to that core function.

Reply:
Was a misunderstanding because it's not useCurrentRoute from the context, but actually "getCurrentRoute" from the conductor.